lib/Parse/ParseDecl.cpp

```diff
@@ -2005,13 +2005,23 @@ ParserStatus Parser::parseDecl(SmallVectorImpl<Decl*> &Entries,
                                                   false);
         }
       }
+      if (CurDeclContext) {
+        auto nominal =
+          CurDeclContext->getAsNominalTypeOrNominalTypeExtensionContext();
+        if (nominal) {
+          diagnose(nominal->getLoc(), diag::note_in_decl_extension, false,
```

Comment:
We can't put notes before the primary warning/error; they'll be associated with the previous diagnostic instead of the next one. Just put it afterwards, it's fine.